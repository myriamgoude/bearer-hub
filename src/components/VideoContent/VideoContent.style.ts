import { css } from '@emotion/core'
import { colors, breakpoints } from '../../styles/variables'

export default {
  root: css`
    margin: auto;
    border-radius: 12px;
    background-color: ${colors.darkBlue};
    background-size: cover;
    box-shadow: 0 8px 17px 0 rgba(3, 13, 54, 0.16);
    position: relative;
    z-index: 5;
    width: 100%;
    height: 389px;
    @media (max-width: ${breakpoints.xl}px) {
      height: 326px;
    }
  `
}
