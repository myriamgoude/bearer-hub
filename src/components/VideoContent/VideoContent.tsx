import * as React from 'react'
import styles from './VideoContent.style'

interface IVideoContentProps {
  thumbnail: string
  src: string
  type?: string
  size?: { width: number; height: number }
}

const VideoContent: React.SFC<IVideoContentProps> = ({
  thumbnail,
  src,
  type = 'video/mp4',
  size = { width: 560, height: 315 }
}) => {
  return (
    <video
      autoPlay={true}
      controls={false}
      loop={true}
      muted={true}
      playsInline={true}
      poster={thumbnail}
      preload={'auto'}
      css={[styles.root]}
      width={`${size.width}px`}
      height={`${size.height}px`}
    >
      <source src={src} type={type} />
    </video>
  )
}
export default VideoContent
