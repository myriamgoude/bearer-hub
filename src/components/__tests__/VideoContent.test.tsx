import * as React from 'react'
import renderer from 'react-test-renderer'
import VideoContent from '../VideoContent/VideoContent'

describe('VideoContent', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<VideoContent src="/my-video.mp4" thumbnail="/mv-video-perview.jpg" />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
