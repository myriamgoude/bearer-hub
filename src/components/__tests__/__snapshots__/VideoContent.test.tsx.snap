// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoContent renders correctly 1`] = `
<video
  autoPlay={true}
  controls={false}
  loop={true}
  muted={true}
  playsInline={true}
  poster="/mv-video-perview.jpg"
  preload="auto"
  src="/my-video.mp4"
  type="video/mp4"
/>
`;
